File: EduCenter_BE/src/main/java/com/example/EduCenter_BE/service/interfaces/ScoreService.java
package com.example.EduCenter_BE.service.interfaces;

import com.example.EduCenter_BE.request.score.CreateScoreRequest;
import com.example.EduCenter_BE.request.score.UpdateScoreRequest;
import com.example.EduCenter_BE.response.ScoreResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface ScoreService {

    ScoreResponse createScore(CreateScoreRequest request);

    Page<ScoreResponse> getAllScores(Pageable pageable);

    ScoreResponse getScoreById(Long id);

    List<ScoreResponse> getScoresByStudentCode(String studentCode);

    Page<ScoreResponse> getScoresByClassroom(String classroomName, Pageable pageable);

    ScoreResponse updateScore(Long id, UpdateScoreRequest request);

    String deleteScore(Long id);
}

File: EduCenter_FE/src/services/score.service.ts
import api from "../api/axios";
import type { Score } from "../types/Score";

export function getAllScores(page = 0, size = 10) {
    return api
        .get("/admin/get-all-scores", { params: { page, size } })
        .then((res) => res.data);
}

export function getScoresByClassroom(
    classroomName: string,
    page = 0,
    size = 10,
) {
    return api
        .get(`/admin/get-scores-by-classroom/${classroomName}`, {
            params: { page, size },
        })
        .then((res) => res.data);
}

export function getScoresByStudent(studentCode: string) {
    return api
        .get(`/admin/get-scores-by-student/${studentCode}`)
        .then((res) => res.data);
}

export function getScoreById(id: number) {
    return api.get(`/admin/get-score/${id}`).then((res) => res.data);
}

export function createScore(data: Partial<Score>) {
    return api.post("/admin/create-score", data);
}

export function updateScore(id: number, data: Partial<Score>) {
    return api.put(`/admin/update-score/${id}`, data);
}

export function deleteScore(id: number) {
    return api.delete(`/admin/delete-score/${id}`);
}
